### PR TITLE
WIP - 3.next - Allow easily generating browser's native date/time inputs

### PR DESCRIPTION
**Pull request by ADmad**
Refs #12817

### CODE REVIEW

**Comment by ADmad** [comment]
Needs tests with values.

`datetime-local` needs values in form `yyyy-MM-ddThh:mm` so we probably need to add a widget for it to format the value as such.

**Comment by ADmad** [comment]
This is ready for some feedback (Ignore missing/incorrect docblocks).

File: src/View/Widget/DateTimeNativeWidget.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+use Exception;
+use RuntimeException;
+
+/**
+ * Input widget class for generating a date time input widget.
+ *
+ * This class is intended as an internal implementation detail
+ * of Cake\View\Helper\FormHelper and is not intended for direct use.
+ */
+class DateTimeNativeWidget implements WidgetInterface
```
Comment (markstory) [inline src/View/Widget/DateTimeNativeWidget.php]: I wonder if we should just make native date inputs the default for 4.0. HTML5 date input support has gotten much better with evergreen browsers and not having to support multiple selectboxes would let us save some complexity internally as well.

Comment (ADmad) [reply src/View/Widget/DateTimeNativeWidget.php]: I am fine with making native datetime inputs default in 4.0 and dropping the selects altogether. In practice most sites/apps use JS based pickers anyway for better consistency and usability.

Comment (markstory) [reply src/View/Widget/DateTimeNativeWidget.php]: We should retain the array marshalling for datetimes for backwards compatibility.  I bet somewhere someone is using array dates in an API.

Comment (ADmad) [reply src/View/Widget/DateTimeNativeWidget.php]: Keeping datetime array marshaling sounds reasonable.

So should I reopen this against `4.x` instead, with the helper and widget code for selects removed?

Comment (markstory) [reply src/View/Widget/DateTimeNativeWidget.php]: I think that is a good idea. We could move the old widgets into a plugin that folks can use if they really need that behavior.

Comment (ADmad) [reply src/View/Widget/DateTimeNativeWidget.php]: Okay, closing this then. Will open a new PR targeting 4.x soon.

Comment (markstory) [reply src/View/Widget/DateTimeNativeWidget.php]: What would you have liked to see?

Comment (markusramsak) [reply src/View/Widget/DateTimeNativeWidget.php]: I prefer using dropdown to select the year for birthday, which can allow to select "9999" for the year if someone prefers say his year of birth. Therefore for birthday I like the select dropdown version.

Comment (markstory) [reply src/View/Widget/DateTimeNativeWidget.php]: I think `year`, and `month` inputs could stay as select boxes. The native HTML inputs are better for `date` and `time` values though.

Comment (ADmad) [reply src/View/Widget/DateTimeNativeWidget.php]: If we are going to switch to native inputs then by default they should be used for all available types. There's no HTML input for `year` so keeping the select box for `year` is fine but there is one for `month` so for that native input should be used.

**Pull request closed without merging**